runtime/bindings/python/iree/runtime/scripts/iree_devices/__main__.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import argparse
+import sys
+
+from typing import Optional, Sequence
+
+from iree import runtime as rt
+
+
+def main(raw_args: Optional[Sequence[str]] = None) -> int:
+  args = create_arg_parser().parse_args(
+      raw_args if raw_args is not None else sys.argv[1:])
+
+  # Switch to sub-command.
+  if args.command == "ls":
+    return main_ls(args)
+  elif args.command == "test":
+    return main_test(args)
+  else:
+    raise RuntimeError(f"Unhandled cli command: {args.command}")
```

Comment:
I see the value in writing this in Python and using the Python APIs, but I would also appreciate the tool appearing in the native C/C++ build and tools installation too. The "console scripts" blur the line between python and C/C++, and I've been finding myself digging through python wheels and directory trees trying to find the underlying binaries from time to time :/

I've written similar "list all devices" code into `iree-run-module` before, so I know that I would find that functionality useful in native tools. I think ideally we'd have both a tool/script like this that enumerates all the options on demand, and some error messages when trying to create a driver/device saying "you asked for X, but I only have [Y, Z]".

---

Maybe it's just my setup, but I also haven't been able to build the Python bindings on Windows:
```
[build] FAILED: compiler/src/iree/compiler/API/python/CMakeFiles/IREECompilerAggregateCAPI.dir/__/__/__/__/__/__/third_party/llvm-project/llvm/resources/windows_version_resource.rc.res 
[build] C:/PROGRA~1/CMake/bin/cmcldeps.exe RC D:\dev\projects\iree\third_party\llvm-project\llvm\resources\windows_version_resource.rc compiler\src\iree\compiler\API\python\CMakeFiles\IREECompilerAggregateCAPI.dir\__\__\__\__\__\__\third_party\llvm-project\llvm\resources\windows_version_resource.rc.res.d compiler\src\iree\compiler\API\python\CMakeFiles\IREECompilerAggregateCAPI.dir\__\__\__\__\__\__\third_party\llvm-project\llvm\resources\windows_version_resource.rc.res "Note: including file: " "C:/Program Files (x86)/Microsoft Visual Studio/2022/Preview/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64/cl.exe" C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe -DCPUINFO_SUPPORTED_PLATFORM=1 -DGTEST_HAS_RTTI=0 -DIREE_HAL_HAVE_VMVX_DRIVER_MODULE=1 -DIREE_HAL_HAVE_VMVX_MODULE_LOADER=1 -DIREE_HAVE_EMITC_DIALECT -DIREE_HAVE_TORCH_MLIR_DIALECTS -DMLIR_PYTHON_PACKAGE_PREFIX=iree.compiler. -D_HAS_EXCEPTIONS=0 -DRC_FILE_VERSION=\"\" -DRC_INTERNAL_NAME=\"IREECompilerAggregateCAPI\" -DRC_PRODUCT_NAME=\"LLVM\" -DRC_PRODUCT_VERSION=\"\" -DRC_VERSION_FIELD_1="" -DRC_VERSION_FIELD_2="" -DRC_VERSION_FIELD_3="" -DRC_VERSION_FIELD_4=0 -I D:\dev\projects\iree\llvm-external-projects\iree-dialects\include -I D:\dev\projects\iree\third_party\llvm-project\mlir\include -I D:\dev\projects\iree\third_party\llvm-project\llvm\include -I D:\dev\projects\iree-build\third_party\llvm-project\llvm\include -I D:\dev\projects\iree-build\third_party\llvm-project\llvm\tools\mlir\include -I D:\dev\projects\iree\third_party\llvm-project\lld\include -I D:\dev\projects\iree-build\third_party\llvm-project\llvm\tools\lld\include -I D:\dev\projects\iree-build\third_party\llvm-project\llvm\tools\mlir-hlo\include -I D:\dev\projects\iree\third_party\cpuinfo\include -I D:\dev\projects\iree\third_party\llvm-project\llvm\..\mlir\include -I D:\dev\projects\iree-build\third_party\llvm-project\llvm\tools\mlir-iree-dialects\include -I D:\dev\projects\iree\third_party\torch-mlir-dialects\include -I D:\dev\projects\iree-build\third_party\llvm-project\llvm\tools\torch-mlir-dialects\include -I D:\dev\projects\iree\third_party\spirv_cross -I D:\dev\projects\iree\compiler\src -I D:\dev\projects\iree\third_party\mlir-hlo\include -I D:\dev\projects\iree -I D:\dev\projects\iree-build -I D:\dev\projects\iree-build\compiler\src -I D:\dev\projects\iree-build\runtime\src\iree\base\internal\flatcc -I D:\dev\projects\iree\third_party\flatcc\include -I D:\dev\projects\iree\runtime\src -I D:\dev\projects\iree-build\runtime\src -I D:\dev\projects\iree-build\runtime\src\iree\schemas -I D:\dev\projects\iree\third_party\mlir-hlo -I D:\dev\projects\iree-build\third_party\mlir-hlo -I D:\dev\projects\iree-build\third_party\mlir-hlo\include -DWIN32 -ID:/dev/projects/iree/third_party/flatcc/include/ -ID:/dev/projects/iree/third_party/flatcc/include/flatcc/reflection/ /DWIN32_LEAN_AND_MEAN /DNOMINMAX /D_USE_MATH_DEFINES /D_CRT_SECURE_NO_WARNINGS /D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES /nologo /fo compiler\src\iree\compiler\API\python\CMakeFiles\IREECompilerAggregateCAPI.dir\__\__\__\__\__\__\third_party\llvm-project\llvm\resources\windows_version_resource.rc.res D:\dev\projects\iree\third_party\llvm-project\llvm\resources\windows_version_resource.rc
[build] 
[build] D:\dev\projects\iree\third_party\llvm-project\llvm\resources\windows_version_resource.rc(56) : error RC2127 : version WORDs separated by commas expected
```

Reply:
The console scripts are for users, and they are pretty good at that.

Idk about Windows compatibility. It used to work.